### PR TITLE
ci: Use clang-15 to build on CI

### PR DESCRIPTION
**Pull request by alimpfard**
Following the serenity CI update, switch to using clang-15.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,9 +43,17 @@ jobs:
           version: 1.10.0
           dest: ${{ github.workspace }}/ninja_bin
 
+      - name: Install Linux dependencies
+        if: ${{ matrix.platform == 'ubuntu-22.04' }}
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
```
Comment (ADKaster) [inline .github/workflows/ci.yml:49]: Using apt-key add like this is deprecated, this should be something like

```
          . /etc/lsb-release
          sudo wget -O /usr/share/keyrings/llvm-snapshot.gpg.key https://apt.llvm.org/llvm-snapshot.gpg.key
          sudo echo "deb [signed-by=/usr/share/keyrings/llvm-snapshot.gpg.key] http://apt.llvm.org/${DISTRIB_CODENAME}/ llvm-toolchain-${DISTRIB_CODENAME} main" | sudo tee -a /etc/apt/sources.list.d/llvm.list
          sudo echo "deb [signed-by=/usr/share/keyrings/llvm-snapshot.gpg.key] http://apt.llvm.org/${DISTRIB_CODENAME}/ llvm-toolchain-${DISTRIB_CODENAME}-15 main" | sudo tee -a /etc/apt/sources.list.d/llvm.list
```

Comment (alimpfard) [reply .github/workflows/ci.yml:49]: :shrug: that's a copy of what we have in serenity
switch over there and I'll copy it here :eyes:

Comment (ADKaster) [reply .github/workflows/ci.yml:49]: I mean... We don't *have* to propagate a "bad" pattern, but if that's how you wanna play it sure, I can update them all at the same time

Comment (alimpfard) [reply .github/workflows/ci.yml:49]: I'll update this one in this PR then, awaiting your fix for serenity~

**Comment by alimpfard** [comment]
Update: Plus one _deflake_ for windows CI, yay!

**Pull request merged**